src/components/Page/components/Header/tests/Header.test.tsx

```diff
@@ -98,6 +98,16 @@ describe('<Header />', () => {
       const expectedButton = buttonsFrom(primaryAction, {primary: false});
       expect(header.contains(expectedButton)).toBeTruthy();
     });
+
+    it('renders a `ReactNode`', () => {
+      const primaryAction = <div>Hello</div>;
+
+      const header = mountWithAppProvider(
+        <Header {...mockProps} primaryAction={primaryAction} />,
+      );
+
+      expect(header.contains(primaryAction)).toBeTruthy();
+    });
```

Comment:
Another cool way this could be handled is using `react-testing` matchers!

```tsx
const PrimaryAction = () => null;

const header = mountWithApp(
  <Header {...mockProps} primaryAction={<PrimaryAction />} />,
);

expect(header).toContainReactComponent(PrimaryAction);
```